src/components/ConnectToQuickbooksOnlineButton/index.native.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useCallback, useRef, useState} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {WebViewNavigation} from 'react-native-webview';
+import {WebView} from 'react-native-webview';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import Button from '@components/Button';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import Modal from '@components/Modal';
+import useLocalize from '@hooks/useLocalize';
+import {getQuickBooksOnlineSetupLink} from '@libs/actions/connections/QuickBooksOnline';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Session} from '@src/types/onyx';
+import type {ConnectToQuickbooksOnlineButtonProps} from './types';
+
+type ConnectToQuickbooksOnlineButtonOnyxProps = {
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+const renderLoading = () => <FullScreenLoadingIndicator />;
+
+function ConnectToQuickbooksOnlineButton({policyID, session}: ConnectToQuickbooksOnlineButtonProps & ConnectToQuickbooksOnlineButtonOnyxProps) {
+    const [isWebViewOpen, setWebViewOpen] = useState<boolean>(false);
+    const [isQuickbooksOnlineReady, setIsQuickbooksOnlineReady] = useState<boolean>(false);
+    const {translate} = useLocalize();
+    const webViewRef = useRef<WebView>(null);
+    const authToken = session?.authToken ?? null;
+
+    const handleNavigationStateChange = useCallback(({url, loading}: WebViewNavigation) => {
+        if (loading || !url.startsWith('https://accounts.intuit.com/app/sign-in')) {
+            return;
+        }
+        setIsQuickbooksOnlineReady(true);
+    }, []);
+
+    return (
+        <>
+            <Button
+                onPress={() => setWebViewOpen(true)}
+                text={translate('workspace.accounting.setup')}
+            />
+            {isWebViewOpen && (
```

Comment:
```suggestion
```

Instead of render condition, use `isVisible` to control modal visibility

```js
isVisible={isWebViewOpen}
```

Reply:
Will that keep the `WebView` mounted? 
What about the `FullPageOfflineBlockingView`, do you think that should be placed inside the `Modal`?

Reply:
> Will that keep the WebView mounted?

No, checking this in Flipper the modal content is only rendered when the visibility flag is true
<img width="1608" alt="Screenshot 2024-04-11 at 10 46 12 PM" src="https://github.com/Expensify/App/assets/16493223/6be9af98-273d-4493-a71e-8a0b1b69fe8e">
<img width="1608" alt="Screenshot 2024-04-11 at 10 46 41 PM" src="https://github.com/Expensify/App/assets/16493223/b8b362fe-5f44-448b-869e-7443b2430417">

Reply:
> What about the FullPageOfflineBlockingView, do you think that should be placed inside the Modal?

Yes, placing it outside results in a weird UI. Placing inside makes more sense but it's missing the back button
<img width="572" alt="Screenshot 2024-04-11 at 10 49 46 PM" src="https://github.com/Expensify/App/assets/16493223/44a77020-59c3-4cd9-bb9b-a74562167c8e">

Reply:
Thank you very much for testing and investigating that